x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsStatsQueryExec.java

```diff
@@ -38,7 +38,7 @@ public enum StatsType {
 
     public record Stat(String name, StatsType type, QueryBuilder query) {
         public QueryBuilder filter(QueryBuilder sourceQuery) {
-            return query == null ? sourceQuery : Queries.combine(Queries.Clause.FILTER, asList(sourceQuery, query));
+            return query == null ? sourceQuery : Queries.combine(Queries.Clause.FILTER, asList(sourceQuery, query)).boost(0.0f);
```

Comment:
is it necessary to put an explicit boost given that we combine the queries into a boolean query filter?

Reply:
The combine method does not necessarily transforms both queries into a filter. When the first query is a boolean query, it will be used as the boolean query to use as a first step for the combination - which means that the first query is unchanged to a filter, and thus can have scoring.